src/main/java/prefs/Preferences.java

```diff
@@ -11,122 +17,262 @@
+     */
     public String getLastLoginUsername() {
-        return global.getLastLoginUsername();
+        return sessionConfig.getLastLoginUsername();
     }
 
+    /**
+     * Sets the last login credentials
+     * @param username The username used during the last login
+     * @param password The password used during the last login
+     */
     public void setLastLoginCredentials(String username, String password) {
-        global.setLastLoginCredentials(username, password);
+        sessionConfig.setLastLoginCredentials(username, password);
+        save();
     }
 
+    /**
+     * Retrieves a list of the last open filters
+     */
     public List<String> getLastOpenFilters() {
-        return global.getLastOpenFilters();
+        return sessionConfig.getLastOpenFilters();
     }
 
+    /**
+     * Retrieves a list of panel names
+     */
     public List<String> getPanelNames() {
-        return global.getPanelNames();
+        return sessionConfig.getPanelNames();
     }
-    
+
+    /**
+     * Retrieves a list of panel infos
+     */
     public List<PanelInfo> getPanelInfo() {
-        return global.getPanelInfo();
-    }
-    
-    public void setPanelInfo(List<PanelInfo> panelInfo) {
-        global.setPanelInfo(panelInfo);
+        return sessionConfig.getPanelInfo();
     }
 
     /**
-     * Interface to configuration files
+     * Sets the panel infos
+     * @param panelInfo The list of panel infos to set to
      */
+    public void setPanelInfo(List<PanelInfo> panelInfo) {
+        sessionConfig.setPanelInfo(panelInfo);
+        save();
+    }
 
     /**
-     * Boards
+     * Adds a board with the board name and its panels information
      */
-
     public void addBoard(String name, List<PanelInfo> panels) {
         assert name != null && panels != null;
-        global.addBoard(name, panels);
+        sessionConfig.addBoard(name, panels);
+        save();
     }
 
+    /**
+     * Retrieves a map of all the boards
+     */
     public Map<String, List<PanelInfo>> getAllBoards() {
-        return global.getAllBoards();
+        return sessionConfig.getAllBoards();
     }
 
+    /**
+     * Retrieves a list of all the board names
+     */
     public List<String> getAllBoardNames() {
         return new ArrayList<>(getAllBoards().keySet());
     }
 
+    /**
+     * Removes the board specified
+     * @param name The name of the board to remove
+     */
     public void removeBoard(String name) {
-        global.removeBoard(name);
+        sessionConfig.removeBoard(name);
+        save();
     }
-    
+
+    /**
+     * Sets the last open board
+     * @param board The name of the last open board.
+     */
     public void setLastOpenBoard(String board) {
-        global.setLastOpenBoard(board);
+        sessionConfig.setLastOpenBoard(board);
+        save();
     }
-    
+
+    /**
+     * Retrieves the name of the last open board.
+     * @return An Optional of a nullable board name
```

Comment:
Remove `nullable`. I think that's implementation specific, and such check should not be in this class (should be in `sessionConfig` class).